### ISSUE TITLE
Does using ImDrawList.Add* disregard window boundaries?

### ISSUE THREAD

**Issue by theyamo** [ocornut/imgui#802]
So my program will use ImGui widgets for top level controls but a section of the screen will be reserved for a custom UI. For convenience I am using AddText and several other methods from ImDrawList for rendering into the custom UI window.

It seems I have to add the actual position of the custom window (within the whole canvas) for each Add* call to make text/lines/rectangles appear in the right place.

So apparently the actual position for the window where text etc should be drawn is calculated somewhere higher in the ImGui drawing hierarchy, not in Add* functions?

I'm fine with doing the position calculations manually in my program but I just hope to confirm that this is intended behaviour and I'm not doing something wrong.

**Comment by ocornut** [comment]
Hello,

Positions passed to the ImDrawList api are indeed always absolute coordinates in screen/display space. 

When using ImGui it maintains absolute screen positions. You can use ImGui::GetWindowPos() or ImGui::GetCursorScreenPos() to get those.
 
Note that the ImDrawList state also hold a clipping rectangle (4 floats), which you may want to interact with if you want to draw outside window boundaries.

Omar

**Comment by ocornut** [comment]
TL;DR; you are correct.

**Comment by theyamo** [comment]
Thanks! ImGui is great by the way ;)
